rfcs/0010-txpool-dos-mitigation.md

```diff
@@ -77,24 +77,58 @@ the transaction and do not gossip it.
     transactions already in the pool. (If it conflicts with one in the pool, see
     below.)
 
-If we get any transactions where the signature is invalid, we blacklist the
-sending node immediately. No honest node sends transactions like that. If we get
-transactions that fail for other reasons - the ones that depend on the current
-chain state - then we increment the node's bad tx count. We track the bad/good
-tx ratio, and blacklist the node if it exceeds `bad_tx_blacklist_ratio`. Honest
-nodes that are out of date might innocently send us transactions that are not
-valid, but they won't send us a lot of them. If a node detects it is
-substantially behind the network, it should disable gossip until it catches up.
-We give new peers `good_faith_tx_credits` good transaction credits when we first
-connect to them. That way we won't erroneously blacklist honest peers early on.
+#### New punishment scheme
```

Comment:
Coda and Bitcoin are basically the same in this respect. If we restrict ourselves to the scenario with discrete bans and not prioritization I think ban scores and trust scores with decay are very close. They differ when a peer is sending lots of transactions, both good and bad. In that case, we could end up banning a peer that sends lots of transactions where a small fraction of them are bad. Imagine an exchange or a payment processor or something that sends 1000tx/min. If 5% of them are bad because of network delay, the node will end up getting banned, which we don't want. But mostly I like this because it's prettier - peers aren't really only good or bad, there are many levels - and it enables the prioritization thing which moves us closer to tit-for-tat.

Reply:
Oops wrong button, didn't mean to resolve.

Come to think of it, the peer sending tons of transactions scenario is much more plausible in Coda since our network TPS should be way higher.